packages/babel-plugin-proposal-decorators/src/transformer-2021-12.ts

```diff
@@ -511,7 +511,7 @@ function transformClass(
       const newField = generateClassProperty(newId, valueNode, isStatic);
 
       const [newPath] = element.replaceWith(newField);
-      addProxyAccessorsFor(newPath, key, newId, element.node.computed);
+      addProxyAccessorsFor(newPath, key, newId, computed);
```

Comment:
This is a bug fix, the `element.node.computed` is of the replacement node, which is always `false`.